front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -258,155 +262,269 @@ const createTrainSchedulePayload = async ({
     };
   });
 
+/**
+ * Generate properties for trainrun related caracteristics.
+ */
+const generateTrainrunProperties = async (
+  netzgrafikDto: NetzgrafikDto,
+  trainrun: TrainrunDto,
+  infraId: number,
+  dispatch: AppDispatch,
+  startTime?: string
```

Comment:
Nit: I'd slightly prefer to take a `Date` here instead of a string, to use stronger types as much as possible.